### ISSUE TITLE
Table with multiple Entity Key columns incorrectly filtering on "Connected To" drill-through

### ISSUE THREAD

**Issue by joshuashort** [metabase/metabase#5648]
I'm using the current master (and other releases).

I've got a table which has 2 Entity Key Fields (it's kind of a bridge between 2 data sets); the fields are `CustID` and `CustNum`.

![screen shot 2017-07-31 at 4 56 34 pm](https://user-images.githubusercontent.com/1498607/28803431-53e63ebc-7611-11e7-9a58-8e542193632a.png)

The "connected to" counts seems to show the correct value, but then they're clicked, the filters are populated with `CustNum` -- some of the relations are on the `CustNum` key, others are on `CustID`.

![screen shot 2017-07-31 at 4 59 31 pm](https://user-images.githubusercontent.com/1498607/28803470-a912b438-7611-11e7-920c-ef1ce4bb4197.png)

In other words, when there are multiple Entity Key(s) defined, it seems like the "connected to" list is correctly finding the relationships, but the links don't apply the correct filter. (in the case of this screenshot, it should be filtering by `CustID = SIR002`

**Comment by salsakran** [comment]
we generally expect that there is only a single entity key in a given record. It is largely analogous to a primary key in sql schema (though we can make use of it even if it isn't indexed/constrained)

From how you're describing it, those fields should be foreign keys.

**Comment by joshuashort** [comment]
I'm basically working with 2 source databases, warehousing them in a single schema. Tables with prefix `vista_` come from our ERP (Epicor Vista 8.03), and tables with prefix `sdb_` come from our Surgical Database.

`vista_Customer`.`CustNum` is the primary key for that table.. but `vista_Customer`.`CustID` also acts like a primary key -- it's user-defined and user-facing though. For example, the table `sdb_Hospitals` uses `CustID` as a foreign key to connect to the ERP data.

The connected-to part is 'working', but it doesn't allow clicking through to see related items -- it seems like it could be made aware of which entity key to pass to filter a table that references it as a foreign key.

**Comment by joshuashort** [comment]
I just realized connected-to isn't quite working like I thought. The relationships using `CustNum` aren't being found, only the ones on `CustID` -- but trying to click through uses `CustNum` for the filter.

**Comment by devurandom** [comment]
Related issues:

* #13798 contains additional information: No screenshots, but more detailed reproduction instructions showing the two problems multiple entity keys create.
* #1177 is related in that it seems to request this feature which appears to exist already.

**Comment by paulrosenzweig** [comment]
Fixed by #15779
